src/app/static/src/app/utils.ts

```diff
@@ -185,6 +186,35 @@ export const flattenOptionsIdsByHierarchy = (filterOptions: NestedFilterOption[]
     return result;
 };
 
+export const flattenXAxisFilterOptionIds = (selections: BarchartSelections, filters: Filter[]) => {
+    const xAxisId = selections?.xAxisId
+    let ids: string[] = []
+    if (xAxisId && filters?.length) {
+        const filter = filters.find((f: Filter) => f.id === xAxisId)
+        if (filter?.options.length) {
+            ids = flattenOptionsIdsByHierarchy(filter.options)
+        }
+    }
+    return ids
+}
+
+
+export const updateSelectionsAndXAxisOrder = (data: BarchartSelections, selections: BarchartSelections, flattenedXAxisFilterOptionIds: string[], updateSelections: (data: {payload: BarchartSelections}) => void) => {
+    const payload = {...selections, ...data}
+    if (data.xAxisId && data.selectedFilterOptions) {
+        const {xAxisId, selectedFilterOptions} = data
+        if (selectedFilterOptions[xAxisId] && flattenedXAxisFilterOptionIds.length) {
+            // Sort the selected filter values according to the order given the barchart filters
+            const updatedFilterOptions = [...selectedFilterOptions[xAxisId]].sort((a: FilterOption, b: FilterOption) => {
+                return flattenedXAxisFilterOptionIds.indexOf(a.id) - flattenedXAxisFilterOptionIds.indexOf(b.id);
+            });
+            payload.selectedFilterOptions[xAxisId] = updatedFilterOptions
+        }
+    }
+    // if unable to do the above, just updates the barchart as normal
+    updateSelections({payload})
+}
```

Comment:
Not sure how necessary it is to test these in the utils file if they are being tested in the component

Reply:
Yeah, we're not very consistent with that sort of thing. I usually do some minimal integrated testing via the component tests, but if there are any edge cases (eg what if `data.xAxisId` is null) then it's usually easier to test those with a direct unit test - though you might want to see how the component would react in those circumstances too. 
If we were doing true unit tests we'd mock the util in the component tests - and that might be worthwhile for some utils which are tricky to set up (eg if they handle file i/o). But it's often useful to test the integration too.